### PR TITLE
UHF-5007 CKEditor Language selector

### PR DESCRIPTION
**Pull request by khalima**
# CKEditor Language selector [UHF-5007](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5007)

A better UX language selector plugin for the CKEditor.

## What was done
* Updated webpack configuration to match with HDBT theme
* Added language selector plugin for the CKEditor and styles for the dialog

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the HDBT Admin theme
    * `composer require drupal/hdbt_admin:dev-UHF-5007_ckeditor_language_selector && composer require drupal/helfi_platform_config:dev-UHF-5007_ckeditor_language_selector`
* Run `make drush-cr drush-updb`

## How to test
* Go to edit any page with CKEditor
* In CKEditor
   * Write a few paragraphs of text and add a phrase in some different language
   * Select the different language part of text and click on the Language selector button next to "Code"
   image
   * Select a language from the dropdown and click ok.
   * The selected part of text should now be in your chosen language
   * Check: https://user-images.githubusercontent.com/1712902/161203000-50706ead-7cd6-41e8-92a5-fb37e2006611.mov
* Save the node (or entity) and check that the translated texts have `<span lang="" dir=""></span>` tags.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/293

### CODE REVIEW

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
+<?php
+
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php:2]: Missing `declare(strict_types = 1)`;

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+class LanguageSelector extends CKEditorPluginBase {
+
+  /**
+   * {@inheritdoc}
+   *
+   * NOTE: The keys of the returned array corresponds to the CKEditor button
+   * names. They are the first argument of the editor.ui.addButton() or
+   * editor.ui.addRichCombo() functions in the plugin.js file.
+   */
+  public function getButtons() {
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php:32]: Missing return type

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        'label' => $this->t('Language selector'),
+        'image' => 'themes/contrib/hdbt_admin/modules/hdbt_admin_editorial/assets/js/plugins/language_selector/icons/language_selector.png',
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFile() {
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php:46]: Missing return type

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    // Make sure that the path to the plugin.js matches the file structure of
+    // the CKEditor plugin you are implementing.
+    return $this->getModuleList()->getPath('hdbt_admin_editorial') . '/assets/js/plugins/language_selector/plugin.js';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isInternal() {
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php:55]: Missing return type

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    return FALSE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDependencies(Editor $editor) {
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php:62]: Missing return type

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLibraries(Editor $editor) {
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php:69]: Missing return type

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    return [
+      'hdbt_admin_editorial/language_selector',
+      'select2_icon/select2_icon',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfig(Editor $editor) {
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/LanguageSelector.php:79]: Missing return type

File: modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/QuoteCKEditorButton.php [hunk 8]
```diff
@@ -43,7 +43,7 @@ public function getButtons() {
   public function getFile() {
```
Comment (tuutti) [inline modules/hdbt_admin_editorial/src/Plugin/CKEditorPlugin/QuoteCKEditorButton.php:43]: Missing return type

**Review by tuutti — CHANGES_REQUESTED** [review]

**Review by tuutti — APPROVED** [review]
Approving PHP changes

**Review by Arkkimaagi — APPROVED** [review]
Works as described, tested with VoiceOver on iOS, works there too. 🚀

**Pull request merged**